data/store/mongo/mongo_datum.go

```diff
@@ -633,13 +589,18 @@ func (d *DatumRepository) GetDataRange(ctx context.Context, userId string, typ s
 	selector := bson.M{
 		"_active": true,
 		"_userId": userId,
-		"type":    typ,
 		"time": bson.M{
 			"$gt":  status.FirstData,
 			"$lte": status.LastData,
 		},
 	}
 
+	if len(typ) > 1 {
```

Comment:
The special case here should be `len(typ) == 1`, because `$in` can handle empty arrays. You should flip the logic even if there is guard earlier in the function. This code was copy/pasted to other functions which do not check if `typ` is non-empty - this wouldn't have been a problem if special case here was handled correctly.

Reply:
done